gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionStore.java

```diff
@@ -30,12 +30,14 @@
 public interface DagActionStore {
   public static final String NO_JOB_NAME_DEFAULT = "";
   enum DagActionType {
+    CANCEL, // Invoked through DagManager if flow has been stuck in Orchestrated state for a while
     KILL, // Kill invoked through API call
-    RESUME, // Resume flow invoked through API call
     LAUNCH, // Launch new flow execution invoked adhoc or through scheduled trigger
+    REEVALUATE, // Re-evaluate what needs to be done upon receipt of a final job status
+    RESUME, // Resume flow invoked through API call
     RETRY, // Invoked through DagManager for flows configured to allow retries
-    CANCEL, // Invoked through DagManager if flow has been stuck in Orchestrated state for a while
-    REEVALUATE // Re-evaluate what needs to be done upon receipt of a final job status
+    ENFORCE_START_DEADLINE, // Enforce job start deadline
+    ENFORCE_FINISH_DEADLINE, // Enforce job finish deadline
```

Comment:
are you sure it's really "job finish"?  somehow I had it in my head that it's "flow completion"